mr/src/main/java/org/elasticsearch/hadoop/util/Version.java

```diff
@@ -52,19 +54,45 @@
 
         if (res != null) {
             List<URL> urls = Collections.list(res);
-            Set<String> normalized = new LinkedHashSet<String>();
+            Map<String, List<URL>> normalized = new LinkedHashMap<String, List<URL>>();
 
+            // On mac, the /tmp dir is a link to /private/tmp due to legacy weirdness.
+            // Both paths get added to the classpath for local YARN node managers when
+            // running Spark. This could happen in any number of other scenarios, and
+            // in each case, the jars that share a canonical path should be identical,
+            // so we'll bucket the URL's by their canonical path before counting them.
             for (URL url : urls) {
-                normalized.add(StringUtils.normalize(url.toString()));
+                try {
+                    String canonicalPath = IOUtils.toCanonicalFilePath(url);
+                    List<URL> pathURLs = normalized.get(canonicalPath);
+                    if (pathURLs == null) {
+                        pathURLs = new ArrayList<URL>();
+                        normalized.put(canonicalPath, pathURLs);
+                    }
+                    pathURLs.add(url);
+                } catch (URISyntaxException e) {
+                    String message = "Could not parse classpath resource URI: " + url.toString();
+                    LogFactory.getLog(Version.class).fatal(message, e);
+                    throw new Error(message, e);
```

Comment:
Why throw `Error` instead of `Exception` or `RuntimeException` ? Error is generally discouraged unless you also own the catching the Error and then only for framework level code as a last resort to prevent crashing the JVM.

Also, any specific reason to log AND throw, usually the thing catching has the responsibility to log as to avoid duplicated logs.